src/test/java/org/jenkinsci/plugins/durabletask/WindowsBatchScriptTest.java

```diff
@@ -98,4 +101,70 @@ private void testWithPath(String path) throws Exception {
         c.cleanup(ws);
     }
 
-}
\ No newline at end of file
+    @Issue("JENKINS-34150")
+    @Test(timeout = 30000) public void batchScriptHanging() throws Exception {
```

Comment:
`timeout` is not needed when using `JenkinsRule`, as it includes an automatic timeout.